### PR TITLE
doc/intro: improve prepositions in "Principle"

### PR DESCRIPTION
**Pull request by zdover23**
This second, later pass over the Introduction (in preparation
for the reorganization of the documentation) makes it possible
for me to see with greater confidence that some of the pre-
positions in this section were not what they should be.

Signed-off-by: Zac Dover <zac.dover@gmail.com>

doc/intro: intro.md - rewrite rest of "Principle"

This PR rewrites the part of the section "Principle" in
the introduction that I did not rewrite in the other pull
reqeust that I made today. After a bit of thinking about
the English grammar that was in the text a few minutes ago,
I decided that it could be made clearer.

Signed-off-by: Zac Dover <zac.dover@gmail.com>

doc/intro: add "sequential write constraint" def.

This adds an explicit identification and definition of the
"sequential write constraint" in the "Principle" section
of the Zoned Storage Devices chapter of the Introduction.

The "sequential write constraint" is referred to very fre-
quently in the ZonedStorage documentation, but there is, as
far as I can recall, no explicit mention of the constraint.

This PR remedies that.

Signed-off-by: Zac Dover <zac.dover@gmail.com>

doc/intro/zoned-storage: rewrite L. eco. support

This PR improves the English in the "Linux Ecosystem Support"
section of the "Zoned Storage" intro.

I have been carefully reading Lydia Davis's notes to her trans-
lation of Proust's "À la recherche du temps perdu", and it has
given me a better sense of prepositions are they are translated
from French to English.

My improved (as compared to last year) understanding of Zoned-
Storage makes me confident that the changes in this PR improve
the readability and accuracy of the English in this section.

Signed-off-by: Zac Dover <zac.dover@gmail.com>

doc/intro: refine "Devel. for ZS"

This PR refines the English in the "Developing for Zoned
Storage" section of the Introduction page.

A year's familiarity with ZonedStorage has increased my confidence
that the changes to the sentences in this PR improve the readability
and technical accuracy of this section.

Signed-off-by: Zac Dover <zac.dover@gmail.com>
(cherry picked from commit d307cc8254a38e5b4dd47a01d833e4ef81f2ab31)

### CODE REVIEW

File: docs/introduction/zoned-storage.md [hunk 1]
```diff
@@ -33,85 +33,88 @@ possible to meet expanding data needs, and SMR technology requires the zoned
...
+   zoned device access constraints from the application scope, implementation
+   of sequential write streams and zone management can be simplified.
+
+3. If **a user lacks software control at the application level but retains
+   control over the choice of the version of the Linux kernel, more advanced
+   kernel support features such as ZBD compliant file systems can be used to
+   hide zoned storage access constraints from the application**. Users may also
+   rely on a device mapper driver that exposes zoned storage devices as regular
+   block devices. **If this solution is used, existing file systems can be used
+   without any modification**.
```
Comment (zdover23) [inline docs/introduction/zoned-storage.md:108]: I want to note here that the emphasized text (that looks like this in the markdown: **text**) is rendered correctly in vim on my local machine, but is not rendered correctly here in github's code browser.

Weird.

Comment (damien-lemoal) [reply docs/introduction/zoned-storage.md:108]: I am not convinced these bold emphasis changes bring any value. And there is a bug in bullet (3) (missing closing **).

Comment (damien-lemoal) [reply docs/introduction/zoned-storage.md:108]: I modified this when applying

**Comment by damien-lemoal** [comment]
Applied manually with modifications. If you squash patches together, please rewrite the commit message into something consistent instead of appending all messages together. And PLEASE, rebase your patches on the latest upstream tree. The patch in this PR contained many changes that were already applied.

**Pull request closed without merging**